### PR TITLE
HOCS-6107: CMS migration document.

### PR DESCRIPTION
**Pull request by rjweeks**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/cms/complaints/ComplaintsService.java [hunk 1]
```diff
@@ -187,6 +192,27 @@ private void extractComplaint(BigDecimal complaintId) {
...
 
+        // create cms migration document
+        try {
+            CaseAttachment caseAttachment = documentCreator.createDocument(complaintId);
+            caseDetails.addCaseAttachment(caseAttachment);
+            ComplaintExtractRecord cmsDocumentStage = getComplaintExtractRecord(complaintId, "CMS data document", true);
+            complaintsRepository.save(cmsDocumentStage);
+        } catch (IOException e) {
+            ComplaintExtractRecord cmsDocumentStage = getComplaintExtractRecord(complaintId, "Couldn't create CMS data document", false);
+            cmsDocumentStage.setError("Couldn't upload PDF");
```
Comment (SWhitlam) [inline src/main/java/uk/gov/digital/ho/hocs/cms/complaints/ComplaintsService.java:203]: Do we want to distinguish the error messages between an IO failure and an extraction failure?

File: src/main/java/uk/gov/digital/ho/hocs/cms/documents/DocumentCreator.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            contentStream.showText(String.format("Representative: %s", representative.getPartyId()));
+            contentStream.setFont(PDType1Font.HELVETICA, fontSize);
+            textForCorrespondent(contentStream, representative);
+        }
+
+        contentStream.endText();
+        contentStream.close();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        document.save(baos);
+        document.close();
```
Comment (SWhitlam) [inline src/main/java/uk/gov/digital/ho/hocs/cms/documents/DocumentCreator.java:121]: Do we have a potential memory leak here if an exception occurs before the document is closed? Maybe use a try with resources and throw a new exception to be caught higher up?

File: src/main/java/uk/gov/digital/ho/hocs/cms/complaints/ComplaintsService.java (hunk 1 above)
Comment (rjweeks) [reply src/main/java/uk/gov/digital/ho/hocs/cms/complaints/ComplaintsService.java]: Added try with resources.

File: src/main/java/uk/gov/digital/ho/hocs/cms/documents/DocumentCreator.java (hunk 2 above)
Comment (rjweeks) [reply src/main/java/uk/gov/digital/ho/hocs/cms/documents/DocumentCreator.java]: Added try with resources on the document.

**Review by SWhitlam — APPROVED** [review]

**Pull request merged**